src/pathfora.js

```diff
@@ -694,6 +705,40 @@
     },
 
     /**
+     * @description check if a customTrigger widget
+     * is ready to be displayed, and if so display the widget
+     * @param {object} widget
+     */
+    triggerWidget: function (widget) {
+      var valid;
+
+      for (var key in widget.watchers) {
```

Comment:
dont worry about it for this pr but seems like some of this global validation logic could be moved to its own function to simplify adding in additional triggers down the road? i haven't looked at the other functions so ignore if its not repeated and if it is lets just make a ticket and address later as a whole.